src/dawn/CodeGen/GridTools/ASTStencilDesc.cpp

```diff
@@ -65,7 +65,53 @@ void ASTStencilDesc::visit(const std::shared_ptr<StencilCallDeclStmt>& stmt) {
 }
 
 void ASTStencilDesc::visit(const std::shared_ptr<BoundaryConditionDeclStmt>& stmt) {
-  DAWN_ASSERT_MSG(0, "BoundaryConditionDeclStmt not yet implemented");
+  if(instantiation_->getBoundaryConditionToExtentsMap().count(stmt) == 0) {
+    DAWN_ASSERT_MSG(false, "Boundary Condition did not trigger extent-calculation");
+  }
+  Extents extents = instantiation_->getBoundaryConditionToExtentsMap().find(stmt)->second;
```

Comment:
This would need some protection, in case the stmt is not found in the map?
And a getter function in stencil instantiation could help make the syntax cleaner here
`Extent getBoundaryConditionExtent(stmt)`